hplsql/src/main/java/org/apache/hive/hplsql/Stmt.java

```diff
@@ -957,7 +957,11 @@ public Integer forCursor(HplsqlParser.For_cursor_stmtContext ctx) {
       int cols = query.columnCount();
       Row row = new Row();
       for (int i = 0; i < cols; i++) {
-        row.addColumnDefinition(query.metadata().columnName(i), query.metadata().columnTypeName(i));
+        String columnName = query.metadata().columnName(i);
+        if (columnName.contains(".")) {
+          columnName = columnName.substring(columnName.lastIndexOf('.') + 1);
```

Comment:
Just try to be a stupid engineer?
How about table like

col1.
col2...
col3.tab1.nice

I would add negative tests as well.

Reply:
@aturoczy we can not create a table or a column with .(dot) in its name so here if .(dot) is there in the name means its _<table_name>.<column_name>_.
```
0: jdbc:hive2://localhost> create table tab.emp(id int, name string);
Error: Error while compiling statement: FAILED: SemanticException [Error 10072]: Database does not exist: tab (state=42000,code=10072)
0: jdbc:hive2://localhost> create table tab(id.new int, name.first string);
Error: Error while compiling statement: FAILED: ParseException line 1:19 cannot recognize input near '.' 'new' 'int' in column type (state=42000,code=40000)
0: jdbc:hive2://localhost>
```